src/models/converters/transfers.rs

```diff
@@ -1,59 +1,175 @@
+    Transfer as TransferDto,
+};
+use crate::models::service::transactions::details::TransactionDetails;
 use crate::models::service::transactions::Transfer as ServiceTransfer;
-use crate::models::backend::transfers::{Transfer as TransferDto, Erc20Transfer as Erc20TransferDto, Erc721Transfer as Erc721TransferDto, EtherTransfer as EtherTransferDto};
+use crate::models::service::transactions::{
+    Erc20Transfer, Erc721Transfer, EtherTransfer, TransactionInfo, TransactionStatus, TransferInfo,
+};
+use crate::providers::info::{
+    InfoProvider, TokenInfo, TokenType,
+};
+use anyhow::Result;
 
 impl TransferDto {
-    pub fn to_transfer(&self) -> TransactionInfo {
+    pub fn to_transfer(&self, info_provider: &mut InfoProvider) -> TransactionInfo {
         match self {
-            TransferDto::Erc721(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            TransferDto::Erc20(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            TransferDto::Ether(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            _ => TransactionInfo::Unknown
+            TransferDto::Erc721(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction(info_provider))
+            }
+            TransferDto::Erc20(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction(info_provider))
+            }
+            TransferDto::Ether(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction())
+            }
+            _ => TransactionInfo::Unknown,
+        }
+    }
+
+    pub fn to_transaction_details(
+        &self,
+        info_provider: &mut InfoProvider,
+    ) -> Result<TransactionDetails> {
+        Ok(TransactionDetails {
+            executed_at: self.get_execution_time(),
+            submitted_at: None,
+            tx_status: TransactionStatus::Success,
+            tx_info: self.to_transfer(info_provider),
+            tx_data: None,
+            tx_hash: self.get_transaction_hash(),
+            detailed_execution_info: None,
+        })
+    }
+
+    fn get_execution_time(&self) -> Option<i64> {
+        match self {
+            TransferDto::Erc721(transfer) => Some(transfer.execution_date.timestamp_millis()),
+            TransferDto::Erc20(transfer) => Some(transfer.execution_date.timestamp_millis()),
+            TransferDto::Ether(transfer) => Some(transfer.execution_date.timestamp_millis()),
+            _ => None,
+        }
+    }
+
+    fn get_transaction_hash(&self) -> Option<String> {
+        match self {
+            TransferDto::Erc721(transfer) => Some(transfer.transaction_hash.to_owned()),
+            TransferDto::Erc20(transfer) => Some(transfer.transaction_hash.to_owned()),
+            TransferDto::Ether(transfer) => Some(transfer.transaction_hash.to_owned()),
+            _ => None,
```

Comment:
Every type appears to contain it as non-nullable, why do we have to make it nullable? Same goes for execution time. Could it be that we should probably shift where this data should go elsewhere?

Reply:
The object was UI focused. On the Ui these are separate sections that can easily be hidden if not present. Therefore the information was grouped this way.